---
title: "Teaching"
---

I have extensive experience in teaching bioinformatic workshops and courses. These have included basic lessons (command line, R, Python, SQL, etc.) as well as more advanced workshops in RNAseq, metagenomics, and population genetics. I am a certified Software Carpentry instructor, and have taught many Software and Data Carpentry lessons.

In my current position I am designing tutorials to help users access and use NEON genetic data, building on the tools in my R package. I am working with colleagues at the National Microbiome Data Collaborative (NMDC) to increase access to these data on the JGI and NMDC data portals. Together we have given workshops on accessing NEON data and I wrote [an R tutorial](https://www.neonscience.org/resources/learning-hub/tutorials/neon-data-metagenomics) to help users begin to integrate the metagenome data across different NEON products.

In my previous position as bioinformatician in the [Department of Anatomy](http://www.otago.ac.nz/anatomy/index.html) at the University of Otago, I ran a bioinformatics help session in the Anatomy department. Most of the material from these sessions is available on the [hacky hour webpage](https://otagomohio.github.io/hacky2021/). These sessions range from getting started writing scripts to some guidelines on troubleshooting command line errors. Over the course of teaching I added some additional reference material to the site. The more advanced workshops generally comprised a combination of live coding and practical problem solving. 

As part of the Otago Carpentries group ([main link](https://otagocarpentries.github.io/)) I have helped put on frequent workshops and training sessions. [Here is a link to past workshops](https://otagocarpentries.github.io/past), which contain a great deal of content. The [Bioinformatics Spring School](https://otagocarpentries.github.io/bioinformatics-spring-school-2020/), held in November, 2020, provides a framework for future large events through Otago Carpentries. It consisted of both basic lessons and specific training in several areas, such as gene expression and population genetics.

I have taught several workshops on environmental DNA (eDNA), and organised an eDNA conference attended by researchers from throughout New Zealand and the Pacific region. As part of the conference, I taught a one-day workshop on metabarcoding, which included an optional day for beginners to learn the basics of the command line and R. I continued to develop the content for this course and taught it several times.

